Tests/GRDBTests/FetchableRecordDecodableTests.swift

```diff
@@ -301,3 +301,200 @@ extension FetchableRecordDecodableTests {
         XCTAssertEqual(value.uuid, uuid)
     }
 }
+
+// MARK: - Custom nested Decodable types - nested saved as JSON
+
+extension FetchableRecordDecodableTests {
```

Comment:
I still don't see any test for detached rows. [Context and guidance](https://github.com/groue/GRDB.swift/pull/397#discussion_r208281803).

Reply:
Start by clicking my "context and guIdance"  link: it starts with a sample code.

@gusrora: If both of you are working for a company and think your manager won't want you to complete this PR since you already have working code for your application, please be totally clear about it as well, so that none of us have unreasonable expectations. We'll say thank you and good bye, and this will be just OK as well.

Reply:
@groue please be reassured that both I and @valexa very much wish to continue on this PR and finish on the open points to so that it is ready to be merged into master branch.

Reply:
@groue have just committed changes to add test for detached rows. Thank you for the example code and the description you gave to help explain detached rows. I have added JSON checking in Data.fromDatabaseValue to allow the reading of Data that is of JSON content. I did not need to add any additions to String.fromDatabaseValue as seems to work fine.

Reply:
> I have just committed changes to add test for detached rows. Thank you for the example code and the description you gave to help explain detached rows.

Great

> I have added JSON checking in Data.fromDatabaseValue to allow the reading of Data that is of JSON content.

Something was not understood at all, here. See this [comment](https://github.com/groue/GRDB.swift/pull/397#issuecomment-410771430) for context.

> I did not need to add any additions to String.fromDatabaseValue as seems to work fine.

I understand, you want to remain focused on your use case and your convenience. But please also understand that this is not enough. I manage this open source library not only for you. If GRDB had been built for my personal use cases only, you would have never heard of it. I repeat: If we support String to Data conversion, we have to support Data to String conversion. It is a matter of consistency. And it was explicitly requested in this [comment](https://github.com/groue/GRDB.swift/pull/397#issuecomment-410771430).

Reply:
This should be done now in 76f5adf9